### PR TITLE
new shaders

### PR DESCRIPTION
**Pull request by norill**
added a bunch of shaders stolen from shadertoy and cool-retro-term

### CODE REVIEW

**Comment by norill** [comment]
requires #19

File: examples/effects/retro/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	y -= 0.5;
+
+	let dist = screenCurvature * (x*x + y*y);
+	dist *= 1.0 + dist;
+
+	return [x * dist + x + 0.5, y * dist + y + 0.5];
+};
+
+return { 
+	passes: [
```
Comment (slammayjammay) [inline examples/effects/retro/index.js]: Anything we can do to reduce the number of passes here? Can we combine some Effects together or remove subtle effects that cause an additional render? It kills my fan...

File: examples/glsl/retro.glsl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+uniform lowp float chromaColor;
+
+uniform lowp float ambientLight;
+
+uniform lowp float horizontalSyncStrength;
+uniform lowp float horizontalSyncFrequency;
+uniform lowp float flickering;
+
+
+// float rgb2grey(vec3 v) {
```
Comment (slammayjammay) [inline examples/glsl/retro.glsl]: Can you remove the commented code unless it serves another purpose?

File: examples/quick-start.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
+const { readFileSync } = require('fs');
+const { resolve } = require('path');
 const pp = require('postprocessing');
 
 module.exports = ({ hyperTerm, xTerm }) => {
-	return { pass: [
+	return { passes: [
```
Comment (slammayjammay) [inline examples/quick-start.js]: Good catch!

File: examples/quick-start.js [hunk 4]
```diff
@@ -25,7 +27,17 @@ module.exports = ({ hyperTerm, xTerm }) => {
 		// new pp.EffectPass(null, new pp.ChromaticAberrationEffect()),
 		// new pp.EffectPass(null, new pp.ColorAverageEffect()),
 		// new pp.EffectPass(null, new pp.BrightnessContrastEffect()),
-		new pp.EffectPass(null, new pp.BloomEffect({ resolutionScale: 0.2, distinction: 0.2 })),
+		// new pp.EffectPass(null, new pp.BloomEffect({ resolutionScale: 0.2, distinction: 0.2 })),
```
Comment (slammayjammay) [inline examples/quick-start.js]: Let's revert the changes made in this array -- it may not work for other users' machines.

**Comment by slammayjammay** [comment]
Ok, holy cow this is absolutely nuts. I looked at that retro shader and it made my day. It looks amazing and it's literally the exact reason this plugin exists, thank you :pray: I want that effect to be the first gif in the demos if that's cool with you. Want to make a gif (I can too)?

Let's get #19 out of the way first and then we'll get this in. Also had some feedback:

The code under examples/ isn't published and mostly serves for demonstrative purposes on how to setup custom effects, so I'd like to keep it as varied as possible. Let's keep the `retro` and `vt220` effects in, but let's remove `mattias-crt` and associated `glsl` files because its two effects (screen curvature and scanlines) are already present in other demos.

Nice work!

File: examples/effects/retro/index.js (hunk 1 above)
Comment (norill) [reply examples/effects/retro/index.js]: oh, i wasn't aware adding more passes degrades performance. in fact i specifically split retro.glsl into parts for greater flexibility. yay for undoing 3 days of work ;(
i think i can merge most of the effects, not sure about bloom though. also burn-in effect needs a save and output pass

File: examples/glsl/retro.glsl (hunk 2 above)
Comment (norill) [reply examples/glsl/retro.glsl]: it was used for alternative burn-in effect, but i liked the other version more and didn't use this one. i commented it out instead of removing for some curious souls to find it and try it. but can remove if you wish

File: examples/quick-start.js (hunk 4 above)
Comment (norill) [reply examples/quick-start.js]: oops, i intended to only add commented out lines, looks like i forgot to uncomment the original line after tests

**Comment by norill** [comment]
yeah, i'm cool with a gif. i already made one, but it's too long and too big. also i'm on windows, so the title bar would be inconsistent with the rest of the gifs. 

there should be a place for people to share their effects. examples folder probably isn't the best place for that, maybe a separate repo? i'd also make effects more self-contained so they are developed and shared  more easily, but i'm getting off topic here

File: examples/effects/retro/index.js (hunk 1 above)
Comment (slammayjammay) [reply examples/effects/retro/index.js]: Whoops I meant Passes, not Effects! Sorry if I caused more work :/. You can have any number of Effects jammed into one Pass, but each additional Pass will hurt performance. I don't really mind how the code is organized as long as the number of passes is minimal.

**Comment by slammayjammay** [comment]
Gif consistency is not a worry but I will make a one later.

Yup I'm open to an examples repo instead of housing them here, and it would be good to make a better way to use/share effects, I just haven't put any time into thinking about it. Feel free to make an issue if you want.

File: examples/effects/retro/index.js (hunk 1 above)
Comment (norill) [reply examples/effects/retro/index.js]: i've gone down to 5 passes. i don't think i can get any lower. i tried combining retro with retro-frame, but the result looked bad. maybe bloom pass can be combined with some other pass, but i don't know how. 

i was looking at cool-retro-termcode and it seems they have shaders divided into 2 parts: static and dynamic. static is computed only once in a while and dynamic is recomputed every frame. it gives a bit more performance, but i'm not sure if it's applicable here

Comment (slammayjammay) [reply examples/effects/retro/index.js]: oh interesting...ok this is fine then, thanks.

File: examples/glsl/retro.glsl (hunk 2 above)
Comment (slammayjammay) [reply examples/glsl/retro.glsl]: i see, let's keep it in then
